Packs/FeedUnit42v2/Integrations/FeedUnit42v2/FeedUnit42v2.py

```diff
@@ -0,0 +1,648 @@
+            for collection in api_root.collections:
+                for bundle in as_pages(collection.get_objects, per_request=100, **kwargs):
+                    data.extend(bundle.get('objects'))
+                    if test:
+                        return data
+
+        self.objects_data[kwargs.get('type')] = data
+
+
+def parse_indicators(indicator_objects: list, feed_tags: list = [], tlp_color: Optional[str] = None) -> list:
+    """Parse the IOC objects retrieved from the feed.
+    Args:
+      indicator_objects: a list of objects containing the indicators.
+      feed_tags: feed tags.
+      tlp_color: Traffic Light Protocol color.
+    Returns:
+        A list of processed indicators.
+    """
+    indicators = []
+    if indicator_objects:
+        for indicator_object in indicator_objects:
+            pattern = indicator_object.get('pattern')
+            for key in UNIT42_TYPES_TO_DEMISTO_TYPES.keys():
+                if pattern.startswith(f'[{key}'):  # retrieve only Demisto indicator types
+                    indicator_obj = {
+                        "value": indicator_object.get('name'),
+                        "type": UNIT42_TYPES_TO_DEMISTO_TYPES[key],
+                        "rawJSON": indicator_object,
+                        "fields": {
+                            "firstseenbysource": indicator_object.get('created'),
+                            "indicatoridentification": indicator_object.get('id'),
+                            "tags": list((set(indicator_object.get('labels'))).union(set(feed_tags))),
+                            "modified": indicator_object.get('modified'),
+                            "reportedby": 'Unit42',
+                        }
+                    }
+
+                    if tlp_color:
+                        indicator_obj['fields']['trafficlightprotocol'] = tlp_color
+
+                    indicators.append(indicator_obj)
+
+    return indicators
+
+
+def get_campaign_from_sub_reports(report_object, id_to_object):
+    report_relationships = []
+    object_refs = report_object.get('object_refs', [])
+    for obj in object_refs:
+        if obj.startswith('report--'):
+            sub_report_obj = id_to_object.get(obj)
+            for sub_report_obj_ref in sub_report_obj.get('object_refs', []):
+                if sub_report_obj_ref.startswith('campaign--'):
+                    related_campaign = id_to_object.get(sub_report_obj_ref)
+
+                    entity_relation = EntityRelationship(name='related-to',
+                                                         entity_a=f"[Unit42 ATOM] {report_object.get('name')}",
+                                                         entity_a_type='Report',
+                                                         entity_b=related_campaign.get('name'),
+                                                         entity_b_type='Campaign')
+                    report_relationships.append(entity_relation.to_indicator())
```

Comment:
not specific to this integration but we should consider `to_indicator` name because we are not really creating an indicator here